File: backend/src/main/java/com/careerguidance/service/AssessmentService.java
package com.careerguidance.service;

import com.careerguidance.dto.request.AssessmentAnswerRequest;
import com.careerguidance.dto.response.AssessmentAnswerResponse;
import com.careerguidance.dto.response.AssessmentCategoryResponse;
import com.careerguidance.dto.response.AssessmentQuestionsByCategoryResponse;
import com.careerguidance.dto.response.AssessmentResultResponse;
import com.careerguidance.dto.response.AssessmentSessionResponse;

import java.util.List;

public interface AssessmentService {
    AssessmentSessionResponse startAssessment(Long userId);

    List<AssessmentCategoryResponse> getCategories();

    List<AssessmentQuestionsByCategoryResponse> getQuestions();

    AssessmentAnswerResponse saveAnswer(Long userId, AssessmentAnswerRequest request);

    AssessmentAnswerResponse updateAnswer(Long userId, Long answerId, AssessmentAnswerRequest request);

    AssessmentResultResponse submitAssessment(Long userId, Long sessionId);

    double calculateTechnicalScore(List<com.careerguidance.entity.AssessmentAnswer> answers);

    double calculateAptitudeScore(List<com.careerguidance.entity.AssessmentAnswer> answers);

    double calculateInterestScore(List<com.careerguidance.entity.AssessmentAnswer> answers);

    String calculatePersonality(List<com.careerguidance.entity.AssessmentAnswer> answers);

    AssessmentResultResponse generateResult(com.careerguidance.entity.AssessmentSession session,
                                            List<com.careerguidance.entity.AssessmentAnswer> answers);

    AssessmentResultResponse getResult(Long userId, Long sessionId);

    List<AssessmentResultResponse> getHistory(Long userId);
}

File: frontend/src/services/assessmentService.js
import api from './api';

const data = request => request.then(response => response.data);

export const assessmentService = {
  startAssessment: () => data(api.post('/api/assessment/start')),
  getQuestions: () => data(api.get('/api/assessment/questions')),
  saveAnswer: payload => data(api.post('/api/assessment/save-answer', payload)),
  updateAnswer: (answerId, payload) => data(api.put(`/api/assessment/update-answer/${answerId}`, payload)),
  submitAssessment: ({ sessionId }) => data(api.post('/api/assessment/submit', { sessionId })),
  getResult: sessionId => data(api.get(`/api/assessment/result/${sessionId}`)),
  getHistory: () => data(api.get('/api/assessment/history')),
};

export const startAssessment = assessmentService.startAssessment;
export const getQuestions = assessmentService.getQuestions;
export const saveAnswer = assessmentService.saveAnswer;
export const submitAssessment = assessmentService.submitAssessment;
export const getResult = assessmentService.getResult;
export default assessmentService;
